src/Factory.php

```diff
@@ -31,6 +31,19 @@ public function publish(string $broker = null): MessageProducer
         );
     }
 
+    /** Creates a new ProducerBuilder instance, setting brokers and topic. The producer will be flushed only when the application terminates, and doing SEND does not mean that the message was flushed! */
+    public function asyncPublish(string $broker = null): MessageProducer
```

Comment:
```suggestion
    public function asyncProducer(string $broker = null): MessageProducer
```

Currently, we use the `producer` method. I think it's better to call this one `asyncProducer` and keep it consistent

Reply:
hmm my mistake, the method is actually called `publish`